File: app/schemas/notification.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from app.models.whatsapp_message_log import MessageLogStatus


class NotificationLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    appointment_id: uuid.UUID | None
    customer_phone: str
    message_type: str
    status: MessageLogStatus
    sent_at: datetime | None
    error_message: str | None
    created_at: datetime


class NotificationLogListResponse(BaseModel):
    items: list[NotificationLogRead]
    total: int
    page: int
    page_size: int

File: crm-frontend/src/types/notification.ts
export type MessageLogStatus = 'pending' | 'sent' | 'failed';

export interface NotificationLog {
  id: string;
  appointment_id: string | null;
  customer_phone: string;
  message_type: string;
  status: MessageLogStatus;
  sent_at: string | null;
  error_message: string | null;
  created_at: string;
}

export interface NotificationLogListResponse {
  items: NotificationLog[];
  total: number;
  page: number;
  page_size: number;
}
